_playwright-tests/UI/helpers/helpers.ts

```diff
@@ -57,6 +57,33 @@ export const getRowCellByHeader = async (row: Locator, name: string) => {
   return row.getByRole('gridcell').nth(index);
 };
 
+export const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
+
+export const waitForTaskPickup = async ({ request }: Page, repoUrl: string, type: string) => {
+  const response = await request.get(`/api/content-sources/v1/repositories/?url=${repoUrl}`);
+  expect(response.status()).toBe(200);
+  const body = await response.json();
+  expect(Array.isArray(body.data)).toBeTruthy();
+  const uuidList = body.data.map((data: { uuid: string }) => data.uuid) as string[];
+  expect(uuidList.length).toEqual(1);
+
+  const repoUuid = uuidList[0];
+  while (true) {
```

Comment:
I am bit torn on this one. It might be necessary for stage/prod tests as we are not in control of queue or for parallel tests, but I feel that this might hide actual issues or make tests unpredictable. The golden rule of tests - if they pass, none cares about them would apply here, as if they pass but always wait for very long time, we would not notice it.

Reply:
I am also a bit torn on this as it's sort of a foot gun for the reason you mentioned, but on the other hand the queue being bogged down on the runner isn't supposed to be the thing this test is verifying so I included this for that reason. 😅 
but I am open to this being axed from this test (or the entire helper) @Andrewgdewar what ya think? 👀